src/v/ssx/abort_source.h

```diff
@@ -20,61 +21,106 @@
 
 namespace ssx {
 
+/**
+ * A sharded version of `ss::abort_source` that allows any shard to request an
+ * abort on all other shards.
+ *
+ * Note that is class's members are wrapped in a shared_ptr. So the object
```

Comment:
Good idea to comment on this but I think it could be a bit clearer, maybe something along the lines of:

```
Sharing one instance of this class across threads raises difficult lifetime and thread safety issues, so the intended usage pattern for this class is to create it on one shard, then make a copy of this original object on every shard where it will be used. All internal state is wrapped in a shared pointer so these copies reference the same underlying state.
```